java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -113,6 +115,16 @@ AllocationListener getListener() {
     return listener;
   }
 
+  @Override
+  public BufferAllocator getParentAllocator() {
+    return parentAllocator;
+  }
+
+  @Override
+  public Collection<BufferAllocator> getChildAllocators() {
+    return childAllocators.keySet().stream().collect(Collectors.toSet());
```

Comment:
Simpler to just do: new HashSet<>(childAlloactors.keySet()) https://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html#HashSet(java.util.Collection)

Reply:
done